src/Storages/NamedCollectionsHelpers.cpp

```diff
@@ -92,14 +92,23 @@ MutableNamedCollectionPtr tryGetNamedCollectionWithOverrides(
     if (asts.size() == 1)
         return collection_copy;
 
+    const auto allow_override_by_default = context->getSettings().allow_named_collection_override_by_default;
+
     for (auto * it = std::next(asts.begin()); it != asts.end(); ++it)
     {
         auto value_override = getKeyValueFromAST(*it, /* fallback_to_ast_value */complex_args != nullptr, context);
 
-        if (!value_override && !(*it)->as<ASTFunction>())
-            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Expected key-value argument or function");
         if (!value_override)
-            continue;
+        {
+            if (!(*it)->as<ASTFunction>())
+                throw Exception(ErrorCodes::BAD_ARGUMENTS, "Expected key-value argument or function");
+            if (allow_override_by_default)
+                continue;
+            // if allow_override_by_default is false we don't allow extra arguments
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Override not allowed");
```

Comment:
It breaks `00002_log_and_exception_messages_formatting`: https://s3.amazonaws.com/clickhouse-test-reports/56171/c0132ea2c709c4c51b47030cb2c15fc7e9243217/stateless_tests__release_.html

Reply:
You should make it more informative

Reply:
For example, the exception message could tell that it's because of disabled `allow_override_by_default` as the comment says

Reply:
It is not skipped in PR's, but does not always fail, may be it is related to parallel running of tests or to division of tests to separate batch jobs.